src/libraries/System.Security.Cryptography.Cose/tests/CoseHeaderMapTests.cs

```diff
@@ -0,0 +1,248 @@
+            map.SetValue(partialIV, ReadOnlySpan<byte>.Empty);
+
+            var writer = new CborWriter();
+            int currentHeader = KnownHeaders.Alg;
+            foreach ((CoseHeaderLabel Label, ReadOnlyMemory<byte> EncodedValue) in map)
+            {
+                if (currentHeader == KnownHeaders.Crit) // TODO remove when Crit gets supported
+                {
+                    currentHeader++;
+                }
+
+                Assert.Equal(new CoseHeaderLabel(currentHeader), Label);
+                ReadOnlyMemory<byte> ExpectedValue = currentHeader switch
+                {
+                    KnownHeaders.Alg => EncodeInt32((int)ECDsaAlgorithm.ES256, writer),
+                    KnownHeaders.Crit => GetDummyCritHeaderValue(),
+                    KnownHeaders.ContentType => EncodeString(ContentTypeDummyValue, writer),
+                    KnownHeaders.Kid => EncodeBytes(s_SampleContent, writer),
+                    KnownHeaders.IV or KnownHeaders.PartialIV => EncodeBytes(ReadOnlySpan<byte>.Empty, writer),
+                    _ => throw new InvalidOperationException()
+                };
+                Assert.True(ExpectedValue.Span.SequenceEqual(EncodedValue.Span));
+                currentHeader++;
+            }
+            Assert.Equal(KnownHeaders.PartialIV + 1, currentHeader);
+
+            static ReadOnlyMemory<byte> EncodeInt32(int value, CborWriter writer)
+            {
+                writer.WriteInt32(value);
+                return EncodeAndReset(writer);
+            }
+
+            static ReadOnlyMemory<byte> EncodeString(string value, CborWriter writer)
+            {
+                writer.WriteTextString(value);
+                return EncodeAndReset(writer);
+            }
+
+            static ReadOnlyMemory<byte> EncodeBytes(ReadOnlySpan<byte> value, CborWriter writer)
+            {
+                writer.WriteByteString(value);
+                return EncodeAndReset(writer);
+            }
+
+            static ReadOnlyMemory<byte> EncodeAndReset(CborWriter writer)
+            {
+                ReadOnlyMemory<byte> encodedValue = writer.Encode();
+                writer.Reset();
+                return encodedValue;
+            }
+        }
+
+        [Fact]
+        public void DecodedProtectedMapShouldBeReadOnly()
+        {
+            byte[] encodedMessage = CoseSign1Message.Sign(s_SampleContent, ECDsaKeys[ECDsaAlgorithm.ES256], HashAlgorithmName.SHA256);
+            CoseSign1Message message = CoseMessage.DecodeSign1(encodedMessage);
+            Assert.True(message.ProtectedHeader.IsReadOnly);
+        }
+
+        [Fact]
+        public void GetValueFromReadOnlyProtectedMap()
+        {
+            byte[] encodedMessage = CoseSign1Message.Sign(s_SampleContent, ECDsaKeys[ECDsaAlgorithm.ES256], HashAlgorithmName.SHA256);
+            CoseSign1Message message = CoseMessage.DecodeSign1(encodedMessage);
+            Assert.True(message.ProtectedHeader.IsReadOnly);
```

Comment:
Assert.True and Assert.False should be labelled, so that if the test fails we have something better than a line number to help identify the failure.

```suggestion
            Assert.True(message.ProtectedHeader.IsReadOnly, "message.ProtectedHeader.IsReadOnly");
```